pyproject.toml

```diff
@@ -152,13 +163,9 @@ legacy_tox_ini = """
         ubuntu-latest: linux
         macos-latest: macos
         windows-latest: windows
-
-    [testenv]
-    commands =
-        pytest --cov --cov-report=lcov
-    deps =
-        pytest-cov
-
-    [tox]
-    env_list = py{310,311,312}-{linux,macos,windows}
 """
+env.testenv = {commands = [
```

Comment:
This does not look right. I think you want here https://tox.wiki/en/4.21.0/config.html#pyproject-toml-native env_run_base, testenv is a INI concept.

Reply:
Yeah, I realised that afterwards (this PR has been reverted). I just tried to do it blind before spotting the docs. I can give it another go now!